openmetadata-ui/src/main/resources/ui/cypress/e2e/Flow/Query.spec.js

```diff
@@ -0,0 +1,149 @@
+/*
+ *  Copyright 2023 Collate.
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+import { visitEntityDetailsPage } from '../../common/common';
+import { SEARCH_ENTITY_TABLE } from '../../constants/constants';
+
+describe('Add Query page', () => {
+  const table = SEARCH_ENTITY_TABLE.table_1;
+
+  const addQuery = ({ inputQuery, expectedErrorMessage }) => {
+    cy.get('[data-testid="add-placeholder-button"]')
+      .should('be.visible')
+      .click();
+
+    cy.get('.ant-form-item-explain-error').should('not.exist');
+
+    inputQuery &&
+      cy
+        .get('.custom-query-editor .CodeMirror-code')
+        .should('be.visible')
+        .type(inputQuery);
+
+    cy.get('[data-testid="save-btn"]')
+      .scrollIntoView()
+      .should('be.visible')
+      .click();
+
+    expectedErrorMessage
+      ? cy
+          .get('.ant-form-item-explain-error')
+          .should('be.visible')
+          .contains(expectedErrorMessage)
+      : cy.get('.Toastify__toast--success').should('be.visible');
+  };
+
+  const deleteQuery = () => {
+    cy.get('[data-testid="queries-container"] .query-card-container')
```

Comment:
if we have data testid then, we can directly use that, no need to class here

Reply:
I have to select the div(query-card-container) that don't have data-testid. Why we should add an extra element data-testid there if I can select it using the already present class?

Reply:
done